### PR TITLE
Switch CI to GitHub Actions

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

**Comment by makaimann** [comment]
Marked `do not merge` because should turn off Travis hooks and delete the `.travis.yml` file before merging.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        name:
+          [
+            btor,
+            cvc4,
+            msat,
+            yices2,
+            z3
```
Comment (yoni206) [inline .github/workflows/ci.yml:15]: no bitwuzla?

File: .travis.yml [hunk 2]
```diff
@@ -75,20 +75,22 @@ jobs:
       env: SOLVER='z3' OPTS=""
 
 script:
+  # disabling Travis for now while evaluating GitHub Actions
```
Comment (yoni206) [inline .travis.yml:78]: maybe consider just removing the file? if we need it later we know where to find it

**Review by yoni206 — APPROVED** [review]
LGTM modulo 2 optional suggestions.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (makaimann) [reply .github/workflows/ci.yml:15]: It's not public yet so there's no good way for me to pull it.

File: .travis.yml (hunk 2 above)
Comment (makaimann) [reply .travis.yml:78]: Yeah I'll definitely delete it before merging, thanks!

**Pull request merged**